### PR TITLE
[Duplicate #5695]Morphological-operations-in-image-processing-using-Open CV

### LINKED ISSUE #5435: Morphological operations in image processing using OpenCV python.
### Proposed title of the article
Morphological operations in image processing using OpenCV python.

### Proposed article introduction
In this tutorial, we will learn about morphological operations. 
What are morphological operations? -These are image processing transformations applied to either grayscale or binary images. These operations require a structuring element, which is used to define the neighborhood of pixels the operation is applied to. how to use morphological operations to increase the size of objects in images as well as decrease them.
We also discuss on  the most important morphological operations that you’ll use inside your own applications:

### Key takeaways
The reader of this article will understand:

The concept of smoothing technique.
The concept of structuring element.
Commonly used operation(s) i.e erosion,dilation 
Edge detection using Canny filter.
How to convert the color image into grayscale shades.
Performing opening and closing operations
Concept of the morphological gradient.
Know the difference between tophat/whitehat and black hat operations. 

### Article quality
This article will not only equip the reader with the knowledge of morphological techniques but also assist him/her with applications of these operations using proper code samples and generating their output.

### References
N/A.

### PR DESCRIPTION
**Pull request by samtecinnovator**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes

This closes #5435

### CODE REVIEW

**Comment by samtecinnovator** [comment]
@WanjaMIKE  i have made the changes requested, you can now proceed with the review thank you.

**Comment by samtecinnovator** [comment]
@hectorkambow  I have made grammatical  edits requested  looking forward for a review.

**Comment by hectorkambow** [comment]
image

**Comment by samtecinnovator** [comment]
requesting for a review

**Comment by hectorkambow** [comment]
@section-engineering-education/peer-reviewers is anyone available to do a prelim review (& tech review) on this. IF ANY more errors persist - we will close asap 👍

File: content/articles/morphological-operations-in-image-processing-using-opencv/index.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+Note that we will use the following image throughout the tutorial:
+[Download-sunflower.png](/engineering-education/morphological-operations-in-image-processing-using-opencv/sunflower-horizon.jpg)
+
+### Prerequisites
+You will require the following to follow along with this tutorial:
+- An IDE which in our case use vs code. You can download vs code[here](https://code.visualstudio.com/download)
+- Download python [Python-download](https://www.python.org/downloads) and set executable path.
+- At the command type `python` then Set up OpenCV to your machine using  ``` pip install cv2 ```.
+- Set up NumPy `pip install NumPy`. 
+- Checking whether or not OpenCV is installed correctly ``` print(cv2.__version__)
```
Comment (WanjaMIKE) [inline content/articles/morphological-operations-in-image-processing-using-opencv/index.md:19]: closing tags `  ```  `

File: content/articles/morphological-operations-in-image-processing-using-opencv/index.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+These operations require a structuring element, that's used to outline the pixels on a picture. 
+
+From this tutorial, users will understand how to apply morphological operations on images.
+
+Note that we will use the following image throughout the tutorial:
+[Download-sunflower.png](/engineering-education/morphological-operations-in-image-processing-using-opencv/sunflower-horizon.jpg)
+
+### Prerequisites
+You will require the following to follow along with this tutorial:
```
Comment (WanjaMIKE) [inline content/articles/morphological-operations-in-image-processing-using-opencv/index.md:14]: Rephrase following to follow

File: content/articles/morphological-operations-in-image-processing-using-opencv/index.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+- An IDE which in our case use vs code. You can download vs code[here](https://code.visualstudio.com/download)
+- Download python [Python-download](https://www.python.org/downloads) and set executable path.
```
Comment (WanjaMIKE) [inline content/articles/morphological-operations-in-image-processing-using-opencv/index.md:16]: Python

File: content/articles/morphological-operations-in-image-processing-using-opencv/index.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+- [Defining structuring element.](#Defining-structuring-element)
+- [Grayscale and rbgcolor images](#Grayscale-and-rbgcolor-images)
+- [Plotting the image](#Plotting-the-image)
+- [Erosion and dilation morphological operations](#Erosion-and-dilation-morphological-operations) 
+- [Edge detection using canny filter](#Edge-detection-using-opencv) 
+- [Canny filter](#Canny-filtering-operations)
+- [Conclusion](#Wrapping-up)
+
+### What is structuring element
+  This is a **matrix** or a **form mask** this is used to identify pixels withinside the image being processed that is crucial in defining neighborhood utilized in every layer of a pixel.
```
Comment (WanjaMIKE) [inline content/articles/morphological-operations-in-image-processing-using-opencv/index.md:35]: that is used ....

File: content/articles/morphological-operations-in-image-processing-using-opencv/index.md [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+### Defining structuring element.
+OpenCV makes use of the subsequent characteristic to outline the structuring element. 
+ ```python 
+ #import important applications 
+ import cv2 
+ #create square strucuring detail of order 10 
+ cv2.getStructuringElement(cv2.MORPH_RECT,(10,10)) 
+ ``` 
+ ###  Grayscale and rbgcolor images 
+  The transformed grayscale picture can also additionally lose contrasts, sharpness, shadow, and shape of the sedation picture.
```
Comment (WanjaMIKE) [inline content/articles/morphological-operations-in-image-processing-using-opencv/index.md:52]: Format well

File: content/articles/morphological-operations-in-image-processing-using-opencv/index.md [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+# Convert the array to make a 200x300 grayscale image.
+grayImage = flatNumpyArray.reshape(300, 200)
+cv2.imwrite('RandomGrayimage.png', grayImage)
+print(grayImage,"\nEND of gray Image\n")
+# Convert the array to make a 200x100 color image.
+bgrImage = flatNumpyArray.reshape(100, 200, 3)
+cv2.imwrite('RandomrgbColor.png', bgrImage)
+#using cv2. imread() it interprets in BGR format
+print(bgrImage,"\nEND of bgr Image")
+  ```
```
Comment (WanjaMIKE) [inline content/articles/morphological-operations-in-image-processing-using-opencv/index.md:74]: What does the above code do?

File: content/articles/morphological-operations-in-image-processing-using-opencv/index.md [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+from matplotlib import image as Img
+# study the picture
+img = Img.imread('content/articles/morphological-operations-in-image-processing-using-opencv/sunflower-horizon.jpg')
+# test if the picture exists
+if (img is None): 
+ print("Could now no longer clear up the picture url")
+else: print(img)
+#plot the binary picture above
+img1plot = plt.imshow(img)
+```
```
Comment (WanjaMIKE) [inline content/articles/morphological-operations-in-image-processing-using-opencv/index.md:96]: What does the above code do?

File: content/articles/morphological-operations-in-image-processing-using-opencv/index.md [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+### Erosion and dilation morphological operations
+**Erosion** is a subset of Euclidean space/integer grid for a few measurements that makes use of a structuring detail for probing and decreasing form contained withinside the  picture its used to shrink linked pixels of 1s in a binary picture.
```
Comment (WanjaMIKE) [inline content/articles/morphological-operations-in-image-processing-using-opencv/index.md:98]: I do not understand this sentence

File: content/articles/morphological-operations-in-image-processing-using-opencv/index.md [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+# dilated image
+print("\n Dilatedimage:\n")
+cv2.imshow(window3, dilatedImg)
+cv2.imwrite('Dilatedimage.jpg', dilatedImg)
+# set wait time to 1 ms
+cv2.waitKey(1)
+# destroy  all sessions and close windows
+cv2.destroyAllWindows()
+
+```
```
Comment (WanjaMIKE) [inline content/articles/morphological-operations-in-image-processing-using-opencv/index.md:148]: What does the above code do?

**Pull request closed without merging**